File: src/main/java/com/instagram/api/repository/UserRepository.java
package com.instagram.api.repository;

import com.instagram.api.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Long> {
    /* username 에 해당하는 회원 존재 여부 */
    boolean existsByUsername(String username);
}

File: kotlin/kotlin-step2/kotlin-app_v2/src/main/kotlin/com/group/libraryapp/domain/user/UserRepository.kt
package com.group.libraryapp.domain.user

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface UserRepository : JpaRepository<User, Long>, UserRepositoryCustom {
//    fun findByName(name: String) : Optional<User>
    fun findByName(name: String) : User?

    // JPA에서 만든 JPQL이 여기 들어가야 한다.
    @Query("SELECT DISTINCT u FROM User u LEFT JOIN FETCH u.userLoanHistoryList")
    fun findAllWithHistoryList(): List<User>
}